textdb/textdb-common/src/main/java/edu/uci/ics/textdb/common/utils/Utils.java

```diff
@@ -72,4 +80,32 @@ public static IndexableField getLuceneField(FieldType fieldType,
         }
         return luceneField;
     }
+    /**
+     * @about Modifies schema, fields and creates a new span tuple
+     */
+    public static ITuple getSpanTuple( List<IField> fieldList, List<Span> spanList, Schema spanSchema) {
+        IField spanListField = new ListField<Span>(new ArrayList<>(spanList));
+        List<IField> fieldListDuplicate = new ArrayList<>(fieldList);
+        fieldListDuplicate.add(spanListField);
+
+        IField[] fieldsDuplicate = fieldListDuplicate.toArray(new IField[fieldListDuplicate.size()]);
+        return new DataTuple(spanSchema, fieldsDuplicate);
+    }
+    
+    /**
+     * 
+     * @param schema 
+     * @about Creating a new schema object, and adding SPAN_LIST_ATTRIBUTE to
+     *        the schema. SPAN_LIST_ATTRIBUTE is of type List
+     */
+    public static Schema createSpanSchema(Schema schema) {
+        List<Attribute> dataTupleAttributes = schema.getAttributes();
+        Attribute[] spanAttributes = new Attribute[dataTupleAttributes.size() + 1];
+        for (int count = 0; count < spanAttributes.length - 1; count++) {

```

Comment:
@chenlica .. We are not completely iterating through `spanAttributes.length`. We want to iterate from 0 to `spanAttributes.length - 2`. So we had put `spanAttributes.length - 1`. @prakul changed it, so that we are using `dataTupleAttributes.size()` which is also fine.
`dataTupleAttributes.size() + 1 = spanAttributes.length `